### PR TITLE
Implemented all function, Updated README

### PR DESCRIPTION
**Pull request by MohamedElgharbawy**
Added double quotes to support pip install on zsh shells (it still works on bash).

### CODE REVIEW

File: tests/numpy/test_misc.py [hunk 1]
```diff
@@ -94,6 +94,15 @@ def test_reshape(nps_app_inst):
     assert nps.allclose(ba.reshape(shape=(6, 4), block_shape=(6, 4)),
                         nps.reshape(ba, shape=(6, 4)))
 
+def test_all(nps_app_inst):
```
Comment (elibol) [inline tests/numpy/test_misc.py:97]: Can you add a test that evaluates bools and floats directly? Something simple is fine -- just make sure you cover both cases, where `all` should evaluate to true and false.

Also, while the test for int is fine (you can do bool the same way), for the float test, test the output against what numpy does. I'm not sure whether it evaluates a float `0.0` as true or false.

Finally, please add an assertion for the dtype of all tests (e.g. `assert nps.all(b).dtype is np.bool`).

**Review by elibol — COMMENTED** [review]
This looks perfect. I left a few comments for the test. Great job!

File: tests/numpy/test_misc.py (hunk 1 above)
Comment (MohamedElgharbawy) [reply tests/numpy/test_misc.py:97]: Added all suggestion.s

Comment (elibol) [reply tests/numpy/test_misc.py:98]: Almost there. Could you add type checks for the bool and int types as you have with int (e.g. `assert nps.all(true_int).dtype is np.bool`)?

Comment (MohamedElgharbawy) [reply tests/numpy/test_misc.py:98]: Done.

**Pull request merged**